Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/ContactDiarySettings.kt

```diff
@@ -1,27 +1,20 @@
 package de.rki.coronawarnapp.contactdiary.ui
 
-import android.content.Context
-import de.rki.coronawarnapp.util.di.AppContext
-import de.rki.coronawarnapp.util.preferences.clearAndNotify
-import de.rki.coronawarnapp.util.preferences.createFlowPreference
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
-class ContactDiarySettings @Inject constructor(
-    @AppContext val context: Context
-) {
+class ContactDiarySettings @Inject constructor(val preferences: ContactDiaryPreferences) {
 
-    private val prefs by lazy {
-        context.getSharedPreferences("contact_diary_localdata", Context.MODE_PRIVATE)
-    }
-
-    val isOnboarded = prefs.createFlowPreference(
-        key = "contact_diary_onboarded",
```

Comment:
This shared preference value is now obsolete. Do we want to remove it?

Reply:
Not a fan, because it creates a dependency to an entity of the domain it is supposed to persist, but if it is you Christmas wish...